### ISSUE TITLE
cx_resolve should always first look in cortex::lang

### ISSUE THREAD

**Issue by SanderMertens** [Seldomberry/Cortex#128]
Currently, the resolve algorithm (implemented in `cx_resolve`) tries first to resolve a name in the provided scope. If no match is found, it will look in the `cortex::lang` scope.

This order is a bit unfortunate, as it would potentially allow a user to redefine a local version of, for example, `class` or `uint32`. This can potentially lead to confusion, and might even be a security concern (someone "spoofing" an object of the type system).

Reversing this order would not only prevent users from using a redefined `class` object. It would also allow referencing to objects in `cortex::lang` by just providing the object name, instead of the fully scoped name (`class` instead of `::cortex::lang::class`). One area in particular where this could make a big difference is in argument lists of functions, which currently are bloated because of the above.

@jleeothon curious what your opinion is

**Comment by jleeothon** [comment]
I understand the tradeoff between resolving one first versus the other. I would think that putting `cortex::lang` first instead of last is not the most expected behavior. Of course, security is an important issue. If a local variable is for some reason created with the name `class`, I think they really want it to be called "class", and users will usually feel discouraged themselves to use such names.

In summary, I'd prefer `::cortex::lang` to be resolved last, but either is fine with me.

I would bring an extra consideration into place, will we plan to include an statement in the kind of C++'s `using`? How is it going to escalate with it?

If so, how does it integrate inside lang? > local > instance > parent scope > lang?

**Comment by SanderMertens** [comment]
This is not related to how the language resolves identifiers. The language has its own resolve algorithm in which local variables are always resolved first. There is no issue there, since you cannot use a variable as a type. The above issue is only relevant to how objects are resolved in the store.

Furthermore, you would still be able to create objects with the name `class` but you would always use a fully scoped name to refer to it. For example:

```Cortex
void Foo::
    void Bar::
        int class: WIDTH_8 // Redefine class in local scope
        class Point::
            x, y: uint32
        
        Foo::Bar::class i: 10 // Uses local class object
```

**Comment by SanderMertens** [comment]
Also, 廌

**Comment by jleeothon** [comment]
Especially taking into account the ability avoid presenting fully qualified names more easily and consistently for proc signatures, ok, that's more than one favorable point. :+1:
